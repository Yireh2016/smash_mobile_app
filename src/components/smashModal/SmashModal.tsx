/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { View, ScrollView, useWindowDimensions } from 'react-native';
import theme from '../../theme/theme';
import Modal from 'react-native-modal';

interface SmashModalProps {
  body: Element;
  children: Element;
  title: Element;
  isModal: boolean;
}

const SmashModal: React.FunctionComponent<SmashModalProps> = ({
  title,
  body,
  children,
  isModal,
}) => {
  const deviceHeight = useWindowDimensions().height;
  return (
    <Modal isVisible={isModal} propagateSwipe={true}>
      <View
        style={{
          backgroundColor: theme.palette.primary,
          padding: 25,
          borderRadius: 10,
          maxHeight: deviceHeight * 0.75,
        }}
      >
        <ScrollView>
          <View>{title}</View>
          <View>{body}</View>
          <View>{children}</View>
        </ScrollView>
      </View>
    </Modal>
  );
};

export default SmashModal;
