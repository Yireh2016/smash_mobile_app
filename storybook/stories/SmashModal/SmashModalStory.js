/* eslint-disable react-native/no-inline-styles */
import React, { useState } from 'react';
import { Text, Button, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import CenterLayout from '../../../src/layouts/centerLayout/CenterLayout';
import SmashModal from '../../../src/components/smashModal/SmashModal';
import { H2, P1 } from '../../../src/components/texts/Texts';

import theme from '../../../src/theme/theme';

import Error from '../../../src/assets/Error';
import SmashButton from '../../../src/components/smashButton/SmashButton';

const CustomModal = ({ isModal, setIsModal }) => {
  const title = <Text>Title</Text>;
  const body = <Text>Body</Text>;
  return (
    <SmashModal title={title} body={body} isModal={isModal}>
      <Button title="ok" onPress={() => setIsModal(false)} />

      <Button title="calcel" onPress={() => setIsModal(false)} />
    </SmashModal>
  );
};

const ErrorModal = ({ isModal, setIsModal }) => {
  const [isVisible, setIsVisible] = useState(false);
  const title = (
    <View>
      <View style={{ flex: 1, alignItems: 'center', marginTop: 50 }}>
        <Error color={theme.palette.error} />
        <View style={{ marginTop: 30 }}>
          <H2>Wait!</H2>
        </View>
      </View>
    </View>
  );
  const body = (
    <View style={{ flex: 1, alignItems: 'center', marginTop: 10 }}>
      <P1>Are you sure you want to unlink this account?</P1>
    </View>
  );
  return (
    <View>
      <Button title="toogle" onPress={() => setIsVisible(!isVisible)} />

      <SmashModal title={title} body={body} isModal={isVisible}>
        <View
          style={{
            marginTop: 25,
            flex: 1,
          }}
        >
          <View
            style={{
              flex: 1,
            }}
          >
            <SmashButton onPress={() => setIsVisible(false)} label="Ok" />
          </View>
          <View
            style={{
              flex: 1,
              marginTop: 20,
            }}
          >
            <SmashButton
              type="secondary"
              onPress={() => setIsVisible(false)}
              label="Cancel"
            />
          </View>
        </View>
      </SmashModal>
    </View>
  );
};

const DefaultModal = () => {
  const [isVisible, setIsVisible] = useState(false);

  return (
    <View>
      <Button title="toogle" onPress={() => setIsVisible(!isVisible)} />

      <CustomModal isModal={isVisible} setIsModal={setIsVisible} />
    </View>
  );
};
export default () =>
  storiesOf('Smash Modal', module)
    .addDecorator((getStory) => (
      <CenterLayout color={theme.palette.primary}>{getStory()}</CenterLayout>
    ))
    .add('Default', () => <DefaultModal />)
    .add('Error', () => <ErrorModal />);
