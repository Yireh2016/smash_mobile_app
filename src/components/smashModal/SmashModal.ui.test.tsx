import * as React from 'react';
import { Text, View } from 'react-native';
import renderer from 'react-test-renderer';
import SmashModal from './SmashModal';

it('SmashModal  renders correctly', () => {
  const ComponentProps = {
    body: <View />,
    children: <View />,
    title: <Text>test</Text>,
    isModal: true,
  };
  const Element = () => {
    return <SmashModal {...ComponentProps} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
