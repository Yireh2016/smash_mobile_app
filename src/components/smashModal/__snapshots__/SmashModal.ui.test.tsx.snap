// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmashModal  renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#10202b",
          "borderRadius": 10,
          "maxHeight": 1000.5,
          "padding": 25,
        }
      }
    >
      <RCTScrollView>
        <View>
          <View>
            <Text>
              test
            </Text>
          </View>
          <View>
            <View />
          </View>
          <View>
            <View />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
